analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -67,6 +67,7 @@ protected override ProgramState[] Process(SymbolicContext context, IInvocationOp
             _ when invocation.TargetMethod.Name == nameof(Equals) => ProcessEquals(context, invocation),
             _ when invocation.TargetMethod.IsAny(KnownType.System_String, nameof(string.IsNullOrEmpty), nameof(string.IsNullOrWhiteSpace)) =>
                 ProcessIsNotNullWhen(state, invocation.WrappedOperation, invocation.Arguments[0].ToArgument(), false, true),
+            _ when invocation.TargetMethod.ContainingType.Is(KnownType.System_String) => ProcessSystemStringInvocation(state, invocation),
```

Comment:
@Tim-Pohlmann in line 66 for ` ProcessLinqEnumerableAndQueryable` see that the current state is being passed (`context.State`) - ignoring any constraints that might have been reset/added from previous lines.

I refactored the method there, in my machine locally, to accept the `state` as parameter and the UTs did not fail :/

If I'm not wrong we should pass also there the `state` and not the `context.State` - and figure out a UT that breaks in case of change. 

Do I miss something here?

Reply:
You miss that this is a switch expression. There is no fall-through. Only one line will get executed.

Reply:
@Tim-Pohlmann I was not clear enough sorry - when I said previous lines I meant the ones before the switch -`33- 58`.

Reply:
Ah... yes!

Reply:
So you agree in passing the state and not the context.State?

Reply:
I added a task for this to my calendar and I'll open an issue if it makes sense.
For now I will leave this PR as is.